File: src/main/java/nl/dannyjelsma/soundboardbot/commands/JoinCommand.java
package nl.dannyjelsma.soundboardbot.commands;

import com.jagrosh.jdautilities.command.Command;
import com.jagrosh.jdautilities.command.CommandEvent;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.GuildVoiceState;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.entities.VoiceChannel;

public class JoinCommand extends Command {

    public JoinCommand() {
        name = "join";
        cooldown = 10;
    }

    @Override
    protected void execute(CommandEvent event) {

        Member member = event.getMember();
        GuildVoiceState voiceState = member.getVoiceState();

        if (voiceState == null || !voiceState.inVoiceChannel()) {
            event.getChannel().sendMessage("You are not in a voice channel...").queue();
            return;
        }

        Guild guild = event.getGuild();
        VoiceChannel channel = voiceState.getChannel();

        guild.getAudioManager().openAudioConnection(channel);
    }
}

File: src/main/java/nl/dannyjelsma/soundboardbot/commands/JoinCommand.kt
package nl.dannyjelsma.soundboardbot.commands

import com.jagrosh.jdautilities.command.Command
import com.jagrosh.jdautilities.command.CommandEvent

class JoinCommand : Command() {
    init {
        name = "join"
        cooldown = 10
    }

    override fun execute(event: CommandEvent) {
        val member = event.member
        val voiceState = member.voiceState

        if (voiceState == null || !voiceState.inVoiceChannel()) {
            event.channel.sendMessage("You are not in a voice channel...").queue()
            return
        }

        val guild = event.guild
        val channel = voiceState.channel
        guild.audioManager.openAudioConnection(channel)
    }
}